apps/wallet/src/ui/app/components/transactions-card/index.tsx

```diff
@@ -46,22 +47,29 @@ export function TransactionCard({ txn, address }: TransactionCardProps) {
     // Get the balance changes for the transaction and the amount
     const balanceChanges = getBalanceChangeSummary(txn, recognizedPackagesList);
 
-    function getAmount(tx: IotaTransactionBlockResponse) {
-        if ((isTimelockedStaking || isTimelockedUnstaking) && tx.events) {
-            return getTransactionAmountForTimelocked(
-                tx.events,
+    const [balance, coinType] = (() => {
+        // Use any non-iota coin type if found, otherwise simply use IOTA
+        const nonIotaCoinType = balanceChanges?.[address]
+            .map((change) => change.coinType)
+            .find((coinType) => coinType !== IOTA_TYPE_ARG);
+        const coinType = nonIotaCoinType ?? IOTA_TYPE_ARG;
+        if ((isTimelockedStaking || isTimelockedUnstaking) && txn.events) {
+            const balance = getTransactionAmountForTimelocked(
+                txn.events,
                 isTimelockedStaking,
                 isTimelockedUnstaking,
             );
+            return [balance, IOTA_TYPE_ARG];
         } else {
-            return address && balanceChanges?.[address]?.[0]?.amount
-                ? Math.abs(Number(balanceChanges?.[address]?.[0]?.amount))
-                : 0;
+            const balanceChange = balanceChanges?.[address]?.find((change) => {
```

Comment:
```suggestion
            const nonIotaCoinType = balanceChanges?.[address]
              ?.map((change) => change.coinType)
              .find((coinType) => coinType !== IOTA_TYPE_ARG);
              const coinType = nonIotaCoinType ?? IOTA_TYPE_ARG;
            const balanceChange = balanceChanges?.[address]?.find((change) => {
```

we use `nonIotaCoinType` `coinType`  values only under `else` condition. I'd move them